src/Adapters/Auth/AuthAdapter.js

```diff
@@ -1,20 +1,102 @@
 /*eslint no-unused-vars: "off"*/
 export class AuthAdapter {
-  /*
+  constructor() {
+    /**
+     * Usage policy
+     * default: can be combined with ONE additional auth provider if additional configured on user
+     * additional: could be only used with a default policy auth provider
+     * solo: Will ignore ALL additional providers if additional configured on user
+     */
+    this.policy = 'default';
+  }
+  /**
   @param appIds: the specified app ids in the configuration
   @param authData: the client provided authData
   @param options: additional options
+  @param req: RestWrite instance with config/auth/data
   @returns a promise that resolves if the applicationId is valid
    */
-  validateAppId(appIds, authData, options) {
+  validateAppId(appIds, authData, options, req) {
+    return Promise.resolve({});
+  }
+
+  /**
+   * Legacy usage, if provided it will be triggered when authData related to this provider is touched (signup/update/login)
+   * otherwise you should implement validateSetup, validateLogin and validateUpdate
+  @param authData: the client provided authData
```

Comment:
Is this valid JSDOc syntax? Shouldn't it be:

`* @param {Object} authData the client provided authData`

Reply:
JSDoc syntax addressed separately